gma/src/ios/ump/consent_info_internal_ios.mm

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gma/src/ios/ump/consent_info_internal_ios.h"
+
+#include "app/src/thread.h"
+#include "app/src/util_ios.h"
+
+namespace firebase {
+namespace gma {
+namespace ump {
+namespace internal {
+
+ConsentInfoInternalIos* ConsentInfoInternalIos::s_instance = nullptr;
+firebase::Mutex ConsentInfoInternalIos::s_instance_mutex;
+
+
+// This explicitly implements the constructor for the outer class,
+// ConsentInfoInternal.
+ConsentInfoInternal* ConsentInfoInternal::CreateInstance() {
+  return new ConsentInfoInternalIos();
+}
+
+ConsentInfoInternalIos::ConsentInfoInternalIos()
+  : loaded_form_(nil) {
+  MutexLock lock(s_instance_mutex);
+  s_instance = this;
+}
+
+ConsentInfoInternalIos::~ConsentInfoInternalIos() {
+  MutexLock lock(s_instance_mutex);
+  s_instance = nullptr;
+}
+
+static ConsentRequestError CppRequestErrorFromIosRequestError(NSInteger code) {
+  switch(code) {
+  case UMPRequestErrorCodeInternal:
+    return kConsentRequestErrorInternal;
+  case UMPRequestErrorCodeInvalidAppID:
+    return kConsentRequestErrorInvalidAppId;
+  case UMPRequestErrorCodeMisconfiguration:
+    return kConsentRequestErrorMisconfiguration;
+  case UMPRequestErrorCodeNetwork:
+    return kConsentRequestErrorNetwork;
+  default:
+    LogWarning("GMA: Unknown UMPRequestErrorCode returned by UMP iOS SDK: %d",
+	       (int)code);
+    return kConsentRequestErrorUnknown;
+  }
+}
+
+static ConsentFormError CppFormErrorFromIosFormError(NSInteger code) {
+  switch(code) {
+  case UMPFormErrorCodeInternal:
+    return kConsentFormErrorInternal;
+  case UMPFormErrorCodeAlreadyUsed:
+    return kConsentFormErrorAlreadyUsed;
+  case UMPFormErrorCodeUnavailable:
+    return kConsentFormErrorUnavailable;
+  case UMPFormErrorCodeTimeout:
+    return kConsentFormErrorTimeout;
+  case UMPFormErrorCodeInvalidViewController:
+    return kConsentFormErrorInvalidOperation;
```

Comment:
Just to double check, is this correct?

Reply:
Yes, we don't have a specific InvalidViewController error code. The error text will specify the issue.